compass/Coordinator.java

```diff
@@ -77,16 +79,44 @@ public Coordinator(CoordinatorConfiguration config, SignatureSource signatureSou
     }).collect(Collectors.toList());
   }
 
+  private static CoordinatorState loadState() throws IOException, ClassNotFoundException {
+    CoordinatorState state;
+    ObjectInputStream ois = new ObjectInputStream(new FileInputStream(CoordinatorState.COORDINATOR_STATE_PATH));
+    state = (CoordinatorState) ois.readObject();
+    ois.close();
```

Comment:
If an exception is thrown we might never the stream object. To mitigate this you can do:
`try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(CoordinatorState.COORDINATOR_STATE_PATH))) {`

This is similar to adding a:
`finally {
  ois.close()
}
`

Reply:
We don't need to deal with the exception in the `loadState`. We want to propagate the exception up the stack and fail. Like it is currently done:
```
     try {
        state = loadState();
      } catch (Exception e) {
        String statePath = System.getProperty("user.dir") + File.separator + CoordinatorState.COORDINATOR_STATE_PATH;
        String msg = "Error loading Compass state file '" + statePath + "'! State file required if not bootstrapping.";

        log.error(msg);
        throw new RuntimeException(msg);
      }
```

Reply:
I think if an error occurs either loading or storing we should bail hard since something is seriously wrong: thus the propagation of exceptions. I am adding a try block for the storeState though, to fail in a more informative way.

Reply:
It is not about the exception...
It is about `ois.close()` which must happen whether the exception will happen or not

You can propagate the exception like you did.

See
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
File descriptors are closed upon process termination, and we always terminate if there is an error in these methods; therefore there is no need to close the fd if we are gonna terminate anyway.

Reply:
That is true,
But still close it.

It is just a best practice :-)

Reply:
Opened an issue https://github.com/iotaledger/compass/issues/55 to not delay the merger of this PR

Reply:
I see your point, making the change